test/controllers/visit.controller.test.js

```diff
@@ -0,0 +1,131 @@
+  });
+
+  context('with valid data', () => {
+    it('returns successful property', (done) => {
+      const booking = VisitFactory.build({ propertyId: propId });
+      request()
+        .post('/api/v1/visit/book')
+        .set('authorization', userToken)
+        .send(booking)
+        .end((err, res) => {
+          expect(res).to.have.status(201);
+          expect(res.body.success).to.be.eql(true);
+          expect(res.body.message).to.be.eql('Visit scheduled successfully');
+          expect(res.body).to.have.property('schedule');
+          done();
+        });
+    });
+  });
+
+  context('when user token is used', () => {
+    beforeEach(async () => {
+      await User.findByIdAndDelete(userId);
+    });
+    it('returns token error', (done) => {
+      const booking = VisitFactory.build({ propertyId: propId });
+      request()
+        .post('/api/v1/visit/book')
+        .set('authorization', userToken)
+        .send(booking)
+        .end((err, res) => {
+          expect(res).to.have.status(404);
+          expect(res.body.success).to.be.eql(false);
+          expect(res.body.message).to.be.eql('Invalid token');
+          done();
+        });
+    });
+  });
+
+  context('when property does not exist', () => {
+    it('returns token error', (done) => {
+      const booking = VisitFactory.build();
+      request()
+        .post('/api/v1/visit/book')
+        .set('authorization', userToken)
+        .send(booking)
+        .end((err, res) => {
+          expect(res).to.have.status(404);
+          expect(res.body.success).to.be.eql(false);
+          expect(res.body.message).to.be.eql('Property not found');
+          done();
+        });
+    });
+  });
+
+  context('with invalid data', () => {
+    context('when property ID is empty', () => {
+      it('returns an error', (done) => {
+        const booking = VisitFactory.build({ propertyId: '' });
+        request()
+          .post('/api/v1/visit/book')
+          .set('authorization', userToken)
+          .send(booking)
+          .end((err, res) => {
+            expect(res).to.have.status(412);
+            expect(res.body.success).to.be.eql(false);
+            expect(res.body.message).to.be.eql('Validation Error');
+            expect(res.body.error).to.be.eql('"Property id" is not allowed to be empty');
+            done();
+          });
+      });
+    });
+    context('when visitor name is empty', () => {
+      it('returns an error', (done) => {
+        const property = VisitFactory.build({ visitorName: '' });
+        request()
+          .post('/api/v1/visit/book')
+          .set('authorization', userToken)
+          .send(property)
+          .end((err, res) => {
+            expect(res).to.have.status(412);
+            expect(res.body.success).to.be.eql(false);
+            expect(res.body.message).to.be.eql('Validation Error');
+            expect(res.body.error).to.be.eql('"Visitor name" is not allowed to be empty');
```

Comment:
"Visitor phone" is not a good response message to users (even though you used Visitor as the field name). Kindly update it and subsequent messages too.
```suggestion
            expect(res.body.error).to.be.eql('"Name" is not allowed to be empty');
```